src/state/messages/convo/agent.ts

```diff
@@ -602,15 +590,6 @@ export class Convo {
       this.isFetchingHistory = true
       this.commit()
 
-      /*
-       * Delay if paginating while scrolled to prevent momentum scrolling from
-       * jerking the list around, plus makes it feel a little more human.
-       */
-      if (this.pastMessages.size > 0) {
-        await new Promise(y => setTimeout(y, 500))
-        // throw new Error('UNCOMMENT TO TEST RETRY')
-      }
```

Comment:
Tested this on sim and doesn't seem to be an issue anymore, nice work @haileyok !